tests/e2e/default/reports/sidebar-filter.wdio-spec.js

```diff
@@ -51,35 +67,63 @@ describe('Reports Sidebar Filter', () => {
       ),
   ];
+  let savedReports;
 
-  beforeEach(async () => {
-    const results = await utils.saveDocs([ parent, patient, ...reports ]);
-    results.splice(0, 2); // Keeping only reports
-    savedReports = results.map(result => result.id);
+  before(async () => {
+    await utils.saveDocs([ ...places.values(), patient ]);
+    savedReports = (await utils.saveDocs(reports)).map(result => result.id);
+    await utils.createUsers([ districtHospitalUser, healthCenterUser ]);
+  });
 
-    await utils.createUsers([ user ]);
-    await loginPage.login(user);
-    await commonElements.waitForPageLoaded();
-    await commonElements.goToReports();
+  afterEach(async () => {
+    await commonPage.logout();
+    await utils.revertSettings(true);
   });
 
   it('should filter by date', async () => {
+    const pregnancyDistrictHospital = savedReports[1];
+    const visitDistrictHospital = savedReports[3];
+
+    await loginPage.login(districtHospitalUser);
+    await commonPage.waitForPageLoaded();
+
+    await commonPage.goToReports();
     await (await reportsPage.firstReport()).waitForDisplayed();
+    expect((await reportsPage.allReports()).length).to.equal(savedReports.length);
+    
     await reportsPage.openSidebarFilter();
-
     await reportsPage.openSidebarFilterDateAccordion();
     await reportsPage.setSidebarFilterFromDate();
     await reportsPage.setSidebarFilterToDate();
+    await commonPage.waitForPageLoaded();
+
+    expect((await reportsPage.allReports()).length).to.equal(2);
+    expect(await (await reportsPage.reportByUUID(pregnancyDistrictHospital)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(visitDistrictHospital)).isDisplayed()).to.be.true;
+  });
 
-    await commonElements.waitForPageLoaded();
-    const allReports = await reportsPage.allReports();
+  it('should filter by user associated place when the permission to default filter is enabled', async () => {
+    const pregnancyHealthCenter = savedReports[0];
+    const visitHealthCenter = savedReports[2];
+
+    await loginPage.login(healthCenterUser);
+    await commonPage.waitForPageLoaded();
+
+    const settings = await utils.getSettings();
+    const newSettings = {
+      permissions: { ...settings.permissions, can_default_facility_filter: [ 'program_officer' ] },
+    };
+    await utils.updateSettings(newSettings);
+
+    await commonPage.goToReports();
+    await (await reportsPage.firstReport()).waitForDisplayed();
 
-    expect(allReports.length).to.equal(2);
-    expect(await (await reportsPage.reportsByUUID(savedReports[1])).length).to.equal(1);
-    expect(await (await reportsPage.reportsByUUID(savedReports[3])).length).to.equal(1);
+    expect((await reportsPage.allReports()).length).to.equal(2);
+    expect(await (await reportsPage.reportByUUID(pregnancyHealthCenter)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(visitHealthCenter)).isDisplayed()).to.be.true;
   });
 });
 
```

Comment:
Great . Thanks for updating this spec 👍 . Also tested in the UI

Reply:
Thank you!